backend/index.ts

```diff
@@ -35,3 +36,53 @@ const port = 8000;
 app.listen(8000, () => {
     console.log(`Server started: http://localhost:${port}/`);
 }); 
+
+
+
+// Configure Strategy for Login
+
+// user is prompted to sign in with their username and password by rendering a form
+// defining a route
+
+app.get('/login/password',
+  function(req, res, next) {
+    res.render('login');
+  });
+
+app.post('/login/password',
+  passport.authenticate('local', { failureRedirect: '/login', failureMessage: true }),
+  function(req, res) {
+    console.log(req.body)
+    res.status(200).send("we logged in")
+  });
+
+var passport = require('passport');
+var LocalStrategy = require('passport-local');
+var crypto = require('crypto');
+var mysql = require('mysql12')
+
+// connecting to the database
+var connection = mysql.createConnection({
+    host: 'localhost',
+    user: 'root',
+    password: '',
+    database: 'wewrap_local'
+  });
+  
+// checking for user info in the database
+passport.use(new LocalStrategy(function verify(username:any, password:any, cb:any) {
```

Comment:
will leave this block to @chinkevin25 to double check

Reply:
we try to avoid using `any` as a type where we can. Do you think we can make our types more specific?

Reply:
also, can we move all of our auth middleware logic into its own folder so we can isolate the index from this login specific logic

Reply:
Done!